src/Entity/OgRole.php

```
@@ -269,6 +268,33 @@ public function isRequired() {
     return static::getRoleTypeByName($this->getName()) === OgRoleInterface::ROLE_TYPE_REQUIRED;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function isAdmin() {
+    return $this->getName() === OgRoleInterface::ADMINISTRATOR;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setIsAdmin($is_admin) {
+    parent::setIsAdmin($is_admin);
+
+    // This method has been inherited from the core RoleInterface but has the
+    // unexpected side effect that it will change the role name. The recommended
+    // way of defining the administrator role is by setting the role name to
+    // OgRoleInterface::ADMINISTRATOR, for example:
+    // @code
+    //   $role->setName(OgRoleInterface::ADMINISTRATOR);
+    // @endcode
+    trigger_error('Using ::setIsAdmin() affects the role name.', E_USER_WARNING);
+
+    $this->setName($is_admin ? OgRoleInterface::ADMINISTRATOR : OgRoleInterface::AUTHENTICATED);
```

This doesn't feel right. We say and `admin` is defined by having a role named `admin`, but if it's not an admin, the role name can be whatever we want.

Maybe we should just throw an Exception, if this method is used?

---

OK, that's also a good solution, alerting the developer using an exception.

@amitaibu to be 100% clear, it is the intention that a group admin has all permissions in the group, right? Because that is wat `::isAdmin()` promises:

```
  /**
   * Indicates that a role has all available permissions.
   *
   * @return bool
   *   TRUE if the role has all permissions.
   */
  public function isAdmin();
```

Reason I am asking is that in my project we have 1 test failing with this PR applied: we have a use case where the group admin is not allowed to delete content, because the content needs to be retained for archival reasons. With the PR the admin gets this permission.

So my understanding is that this is intentional, and on our end we should implement `hook_og_user_access_alter()` to take away the permission from the group admin?

---

ahh, hello :) There is an explicit `is_admin` flag that controls this in core. We can use this. We don't need to restrict it in the administrator role.
The og_roles have inherited this flag and are already in exported configurations. Core also bases this in this flag.

---

Thanks for the input @idimopoulos! I can see the merit of your proposal. It would simplify this PR, and we would stay closer in spirit to the core implementation, while still retaining the full flexibility.

So to be clear, is this what you are suggesting?

* We keep the `is_admin` flag in exported config. When creating a new group type this flag will be automatically set when the admin role is auto-created.
* We keep the core mechanic of determining the admin role solely by this flag, so we will ignore the role name.

Advantages:

* We don't need to override `::isAdmin()` and `::setIsAdmin()` or prevent their usage. They will work as designed.
* The intended behavior remains the same: when a new group type is created, we will have an admin role that has the role name "administrator" and it will have the flag set.
* Developers have the flexibility to create multiple admin roles, just like it can be done in core. I don't see many use cases for this, but there might be.
* Developers have the flexibility to revoke admin rights from the pre-created admin role, just like they can do in core, by calling `$role->setIsAdmin(FALSE)->save()`.

Disadvantages:

* It is possible to have a role that is called "administrator" but is in fact a regular role. This is a minor drawback, it matches the core functionality. It also requires a manual action from a developer to get into this state.